lib/x11utils.pm

```diff
@@ -68,6 +68,17 @@ all possible options should be handled within loop to get unlocked desktop
 =cut
 sub ensure_unlocked_desktop {
     my $counter = 10;
+
+    # check if the screenbuffer was not updated and record a softfailure in that case
+    if (check_var('DESKTOP', 'gnome') && check_screen('screenlock')) {
+        send_key 'shift';
+        wait_still_screen;    # redrawing, turning on guest display or showing animation may take some seconds
```

Comment:
```suggestion
        wait_still_screen 30;    # redrawing, turning on guest display or showing animation may take some seconds
```
This should be more than enough

Reply:
That would be a sleep of time of *at least* 30s, that's the opposite of what I would do

Reply:
> That would be a sleep of time of _at least_ 30s, that's the opposite of what I would do

Then be more helpful and state what you would instead.

Reply:
I would start with 2 seconds and increase it later if the bug is reproduced again and 2 seconds are not enough

Reply:
I am sorry if I was not clear. The problem of `wait_still_screen` is that it can lead to huge waste of time in non-obvious ways. Take a look in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10163 for a recent example of DimStar77 removing one innocent `wait_still_screen 3` and replacing it with a properly designed needle. The result was a speedup of the yast2_control_center test module runtime from 39 minutes to just 3 minutes (!). My common observation is that `wait_screen_change` and `wait_still_screen` is added for two reasons: 1. people are afraid of bad performance due to new needles when it is actually the opposite, 2. people do not know how a proper specific needle check can look like

Here I think it is the second case. But unfortunately I myself did not understand the problem enough to be able to suggest you something more useful.